sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2472,6 +2472,16 @@ object GpuOverrides {
               .withPsNote(TypeEnum.STRING, "only a single character is allowed"), TypeSig.STRING),
           ParamCheck("count", TypeSig.lit(TypeEnum.INT), TypeSig.INT))),
       (in, conf, p, r) => new SubstringIndexMeta(in, conf, p, r)),
+    expr[StringRepeat](
+      "StringRepeat operator",
+      ExprChecks.projectNotLambda(TypeSig.STRING, TypeSig.STRING,
+        Seq(ParamCheck("input", TypeSig.lit(TypeEnum.STRING) + TypeSig.STRING, TypeSig.STRING),
+          ParamCheck("repeatTimes", TypeSig.lit(TypeEnum.INT) + TypeSig.INT, TypeSig.INT))),
```

Comment:
I'm not sure if those checks are correct. I want to check if
 * The first input to this function can either be a string scalar or a strings column.
 * The second input can be an integer number or a numeric column of integer numbers.

Reply:
You can examine the generated documentation for clues as to whether you got the type checking setup correctly.  Also adding unit tests for these cases will further prove this.  Adding `TypeSig.lit` appears to be limiting the support to a scalar, so I'd remove that.  Simply allowing strings for that parameter should cause columnar inputs to call your columnar version of `doColumnar` and a scalar input will call the form that takes a scalar.

Reply:
The `TypeSig.lit(..)` means it is adding a limitation that it only supports a literal input value.  `TypeSig.INT` indicates it can support both a literal and a column as input.  Mixing them I think gives preference to the lit version so this code is saying that `repeatTimes` and `input` both can only support a literal input value, which when Spark sees this it typically runs this during compile and replaces the entire thing with a literal that is the output of running the StringRepeat command. From the code it looks like we can drop the `TypeSig.lit` parts entirely.